### PR TITLE
added icon function

### PR DESCRIPTION
**Pull request by Afanasi0101**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -8,6 +8,6 @@
   </head>
   <body>
     <div id="app"></div>
-    <script type="module" src="/main.js"></script>
+ <script type="module" src="/main.js"></script>
```
Comment (aleksanderwalczuk) [inline index.html:11]: wrong indentation

File: src/components/btn.js [hunk 2]
```diff
@@ -6,15 +6,35 @@ const btnTypes = {
...
+  <path d="M13.06 15.473a48.45 48.45 0 017.666-3.282c.134 1.414.22 2.843.255 4.285a.75.75 0 01-.46.71 47.878 47.878 0 00-8.105 4.342.75.75 0 01-.832 0 47.877 47.877 0 00-8.104-4.342.75.75 0 01-.461-.71c.035-1.442.121-2.87.255-4.286A48.4 48.4 0 016 13.18v1.27a1.5 1.5 0 00-.14 2.508c-.09.38-.222.753-.397 1.11.452.213.901.434 1.346.661a6.729 6.729 0 00.551-1.608 1.5 1.5 0 00.14-2.67v-.645a48.549 48.549 0 013.44 1.668 2.25 2.25 0 002.12 0z" />
+  <path d="M4.462 19.462c.42-.419.753-.89 1-1.394.453.213.902.434 1.347.661a6.743 6.743 0 01-1.286 1.794.75.75 0 11-1.06-1.06z" />
+</svg>
+`,
+  userIcon: `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="w-6 h-6 user_icon">
+  <path fill-rule="evenodd" d="M7.5 6a4.5 4.5 0 119 0 4.5 4.5 0 01-9 0zM3.751 20.105a8.25 8.25 0 0116.498 0 .75.75 0 01-.437.695A18.683 18.683 0 0112 22.5c-2.786 0-5.433-.608-7.812-1.7a.75.75 0 01-.437-.695z" clip-rule="evenodd" />
+</svg>
+`,
+};
```
Comment (aleksanderwalczuk) [inline src/components/btn.js:20]: this is nice, but you put it in wrong place.

File: src/components/btn.js [hunk 3]
```diff
@@ -6,15 +6,35 @@ const btnTypes = {
...
   }
+
+  const boxIcon = document.createElement('span');
+  const typ = iconTypes[iconType];
+  universalBtn.appendChild(boxIcon);
+  boxIcon.innerHTML = `${typ}${text}`;
+  if (iconTypes[iconType] == null || iconTypes[iconType] !== typ) {
+    boxIcon.innerHTML = `${text}`;
+  }
+
```
Comment (aleksanderwalczuk) [inline src/components/btn.js:37]: We should have a separate component for icon creation.

File: src/components/btn.js [hunk 4]
```diff
@@ -6,15 +6,35 @@ const btnTypes = {
   baseLarge: 'btn base_btn_large',
   nav: 'nav',
 };
+const iconTypes = {
+  hatIcon: `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="currentColor" class="w-6 h-6">
```
Comment (aleksanderwalczuk) [inline src/components/btn.js:10]: we already know these are `icons` so we don't have to mention it in the key name.
Later on we'd have to pick it like so: `iconTypes[hatIcon]`. 
Wouldn't it be better to make it `iconTypes[hat]`? 
I think it's obviously better, what about you?

File: src/components/btn.js [hunk 5]
```diff
@@ -6,15 +6,35 @@ const btnTypes = {
...
   const type = btnTypes[btnType];
   const universalBtn = document.createElement('button');
-  universalBtn.textContent = text;
   universalBtn.className = type;
   if (btnTypes[btnType] == null || btnTypes[btnType] in btnTypes) {
     universalBtn.className = btnTypes.base;
   }
+
+  const boxIcon = document.createElement('span');
+  const typ = iconTypes[iconType];
```
Comment (aleksanderwalczuk) [inline src/components/btn.js:31]: what `typ` is?

File: src/scss/components/btn.scss [hunk 6]
```diff
@@ -47,4 +47,14 @@
...
+  width: 37px;
+  height: 37px;
+}
+.user_icon{
+  width: 41.67px;
+  height: 33.33px;
+  color: #DADADA;
+  background-color: $color-black;
+  border-radius: 4px;
+}
```
Comment (aleksanderwalczuk) [inline src/scss/components/btn.scss:60]: This is wrong, we should keep it in `icons.scss`

**Review by aleksanderwalczuk — CHANGES_REQUESTED** [review]
Screenshot 2022-09-24 at 17 02 55

We have icons inside headers too, so we shouldn't add it to the button component.

Beside that, there is a file conflict in `btn` component because you forgot to git pull when you created new branch.

Please remove it make `git pull` on `dev` branch and start again. You can close this PR and make another one if you want to.

File: src/components/btn.js (hunk 4 above)
Comment (Afanasi0101) [reply src/components/btn.js:10]: I think you're right

**Pull request closed without merging**